File: ruby/resistor-color/resistor_color.rb
module ResistorColorDuo
  BANDS = {
    "black"  => 0,
    "brown"  => 1,
    "red"    => 2,
    "orange" => 3,
    "yellow" => 4,
    "green"  => 5,
    "blue"   => 6,
    "violet" => 7,
    "grey"   => 8,
    "white"  => 9
  }
  def self.value(bands)
    bands.take(2).map { |b| BANDS[b] }.join.to_i
  end
end

File: csharp/resistor-color/ResistorColor.cs
using System;

public static class ResistorColor
{
    private static string[] ColorValues = new []
    {
        "black",
        "brown",
        "red",
        "orange",
        "yellow",
        "green",
        "blue",
        "violet",
        "grey",
        "white"
    };
    public static int ColorCode(string color)
    {
        return Array.IndexOf(ColorValues, color);
    }

    public static string[] Colors()
    {
        return ColorValues;
    }
}